### PR TITLE
Updates the pkg/identity support encode and decode functionality of Peer Identity information

### PR DESCRIPTION
**Pull request by aligeti**
What: Updates the Identity packet to support encoding/decoding of the peer Identity to/From bytes, which can be used to save the whole chain of peer identities in database that stores these information. 
Why:
This PR  is a sub task of the a big feature as described in V3-1992 epic. The V3-1992 will be divided into 6 PRs. This is the first of the 6 PRs.

**The 6 PRs would be as shown below:**
1.  make the encode /decode changes  into the pkg/identities ( PR #2754 )
2. remove certdb (because the design changed quite drastically)
3. implement new certdb (dbx and interface and tests)
4. fix those bootstrap error outputting debug info
5. fix the segment debug outputing info (edited) 
6. wire up the new certdb (edited) 

Please describe the tests:
 - Test 1:adds test function to encode
 - Test 2:adds test function to decode
 - Test 3: compares the peer identities by first encodes, then decodes,  and then compares decoded data with the original information used in encoding. 
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/identity/identity.go [hunk 1]
```diff
@@ -527,3 +528,41 @@ func backupPath(path string) string {
 		pathExt,
 	)
 }
+
+// EncodePeerIdentity encodes the complete idenitity chain to bytes
```
Comment (ethanadams) [inline pkg/identity/identity.go]: nit - spelling of identity

File: pkg/identity/identity.go [hunk 2]
```diff
@@ -527,3 +528,41 @@ func backupPath(path string) string {
...
+	var chain []byte
+	chain = append(chain, pi.Leaf.Raw...)
+	chain = append(chain, pi.CA.Raw...)
+	for _, cert := range pi.RestChain {
+		chain = append(chain, cert.Raw...)
+	}
+	return chain
+}
+
+// DecodePeerIdentity Decodes the bytes into complete idenitity chain
```
Comment (ethanadams) [inline pkg/identity/identity.go]: spelling of identity

Comment (aligeti) [reply pkg/identity/identity.go]: Extremely sorry overlooked the comment and checked ... will  fix it ..

File: pkg/identity/identity.go (hunk 1 above)
Comment (aligeti) [reply pkg/identity/identity.go]: Extremely sorry overlooked the comment and checked ... will  fix it ..

**Pull request merged**